packages/muze-utils/src/DataSystem/selection.js

```diff
@@ -105,109 +82,162 @@ class Selection {
      *
      * @param {Function} callback Callback to execute on each item.
      * @return {Selection} New selection with data created using callback.
-     * @memberof EnterSelection
+     * @memberof Selection
      */
     append (callback) {
-        this[`_${this._mode}data`].forEach((...params) => {
-            const data = params[0];
-            const id = this._idGetter ? this._idGetter(...params) : (data.id || params[1]);
-            this._idMap[id] = callback(...params);
-            this._dataMap[id] = data;
-        });
+        let currentData;
+        let val;
+        const dataObjects = new Map();
+        const data = new Map();
+
+        // select the data to create object
+        switch (this._mode) {
```

Comment:
This can be written as `currentData = this[_${mode}data]`

Reply:
decreases code readablity.